### PR TITLE
client folder refactor

### PR DESCRIPTION
**Pull request by Mjavala**
#11 Changes:

- Extracted client folder into its own package from the ``app`` package.
- Refactored app to use the new ``@sunrisestake/client`` package
- Add Lerna and a few extra scripts, linting

TODO:

- Publish client package
- Github actions update

### CODE REVIEW

File: packages/client/dist/blaze.d.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Transaction } from "@solana/web3.js";
```
Comment (dankelleher) [inline packages/client/dist/blaze.d.ts:1]: Make sure to add a gitignore entry for the dist folder

**Review by dankelleher — COMMENTED** [review]
Looks good so far! One comment and just needs CI fixes and everything to go green.

File: packages/client/dist/blaze.d.ts [hunk 2]
```diff
@@ -1,37 +1,40 @@
-import { Transaction } from "@solana/web3.js";
+import { PublicKey, Transaction } from "@solana/web3.js";
```
Comment (dankelleher) [inline packages/client/dist/blaze.d.ts:1]: Dont forget to delete the dist folder from the PR as well as marking as ignored

File: packages/scripts/package.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
     "@project-serum/anchor": "^0.26.0",
     "@solana/web3.js": "^1.66.2",
+    "@sunrisestake/client": "^0.1.0",
```
Comment (dankelleher) [inline packages/scripts/package.json:8]:
```suggestion
    "@sunrisestake/client": "*",
```

File: packages/tests/package.json [hunk 4]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "dependencies": {
     "@solana/web3.js": "^1.66.2",
+    "@sunrisestake/client": "^0.1.0",
```
Comment (dankelleher) [inline packages/tests/package.json:7]:
```suggestion
    "@sunrisestake/client": "*",
```

File: .gitignore [hunk 5]
```diff
@@ -5,3 +5,4 @@ target
 **/*.rs.bk
 node_modules
 test-ledger
+dist/
```
Comment (ronnyhaase) [inline .gitignore:8]: I believe you intend to remove the slash to have dist not version controlled?

File: package.json [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
-        "packages/tests"
+        "packages/tests",
+        "packages/client"
     ],
     "private": true,
     "scripts": {
         "program:lint": "cargo fmt --all -- --check && cargo clippy -- --deny warnings",
         "program:dylint": "cargo dylint --all --workspace",
-        "lint": "eslint .",
+        "lint": "eslint . --fix --ext .ts",
```
Comment (ronnyhaase) [inline package.json:11]: I prefer to separate lint and auto-format, but not strongly opinionated

Comment (dankelleher) [reply package.json:11]: Yeah, this needs to be reverted or CI will try to fix them.

Best to just write `yarn lint --fix` locally. If you like we can add a `yarn lint:fix`

Also if there is consensus we can add husky

File: packages/client/index.ts [hunk 7]
```diff
@@ -63,10 +64,22 @@ import {
   blazeWithdrawSol,
   blazeWithdrawStake,
 } from "./blaze";
-import { ZERO } from "../util";
 import { BlazeState } from "./types/Solblaze";
 import { getStakePoolAccount, StakePool } from "./decode_stake_pool";
 
+// export getSakePoolAccount
```
Comment (ronnyhaase) [inline packages/client/index.ts:70]: Think those comments are unnecessary

File: package.json (hunk 6 above)
Comment (ronnyhaase) [reply package.json:11]: > Yeah, this needs to be reverted or CI will try to fix them.
> 
> Best to just write `yarn lint --fix` locally. If you like we can add a `yarn lint:fix`
> 
> Also if there is consensus we can add husky

100% agree and like

File: packages/client/index.ts (hunk 7 above)
Comment (Mjavala) [reply packages/client/index.ts:70]: yeah, i used them so Copilot knows what i want lol

**Comment by Mjavala** [comment]
Not sure what's going on with vercel, can't access it. @dankelleher

File: package.json (hunk 6 above)
Comment (dankelleher) [reply package.json:11]: Let's do it as a separate PR to avoid overloading this one

**Comment by dankelleher** [comment]
image
Build is failing. Give me your email address (not here) and I'll add you to the vercel group so you can see the issue. You should be able to reproduce it locally with `yarn build` in the app folder.

**Review by dankelleher — APPROVED** [review]

**Pull request merged**